Fix subnet recommendation for staging in Qubes OS

gonzalo-bulnes opened the pull request:
## Status
<!-- What state is your PR in? Select one of the following and delete the option that does not apply. -->

Ready for review


## Description of Changes

- Updates the list of fields that are required by the Ubuntu 20.04 (Focal) installer
- Rename the _hostname_ field as named in the installer, and move it down because it is required in a later step to the network settings
- Modifies the subnet recommendation in for staging in Qubes OS in order to ensure that the resulting machines have access to the network

Fixes #209 
Fixes #211 

## Testing
<!-- How should the reviewer test this PR? Write out any special testing steps here. -->
- [ ] Review the list of fields required by the installer
- [ ] Check that `10.137.0.0/24` indeed contains the likely IPs of both `sd-staging-base-focal` and its gateway

## Release 
<!-- Any special considerations for release of this change into the stable version of the documentation? -->
_No special considerations for release come to my mind._


## Checklist (Optional)

- [x] Doc linting (`make docs-lint`) passed locally
- [ ] Doc link linting (`make docs-linkcheck`) passed
- [x] You have previewed (`make docs`) docs at http://localhost:8000

gonzalo-bulnes commented:
@creviera I'm curious about your network configuration in Qubes OS. Maybe we could review this together :slightly_smiling_face:

conorsch commented:
Hey @gonzalo-bulnes ! This one's been open a whlie, we just dug it up again at sprint-planning earlier today. I offered to take a first pass at it, because I wanted to ask you whether the networking settings were tested on Qubes 4.0 or 4.1. That issude aside, I ended up reading up on some networking settings here and share my thoughts below. 

One of the recommended changes here includes declaring the subnet for the VM are `10.136.0.0/14`, which intuitively strikes me as far too large: 2^18, or 262,144. Turns out I don't know what the proper space is here, so I tried to look it up.

The question I'm trying to answer is "What is the CIDR subnet encompassing IPs for Qubes VMs?"


I've read https://www.qubes-os.org/doc/networking/ and grepped through the code in https://github.com/QubesOS/qubes-core-admin/blob/ee34eb2a936c47c6e20a51d0ac0701929990511d/qubes/vm/mix/net.py#L168-L179 and I don't see any mention of `10.136.*` IPv4 addrs at all! The code does mention `10.137.*` and `10.138.*`, however. Note the bitwise shift operators in the Qubes network code, as well as the commented observation that the AppVM's xen id (qid) is used to determine the IP address. Which is a fine way to ensure we don't have collisions. Translated into a Python one-liner for demonstration purposes:

```
python3 -c 'for i in range(0, 255): print("10.137.{}.{}".format((i >> 8) & 0xff, i & 0xff))'
```

That's the same as:

```
$ netmask -c -r 10.137.0.0/24
     10.137.0.0-10.137.0.255    (256)
```

As an aside, apparently DispVMs use `10.138.0.1/24`, so it's possible to tell whether a VM is Disposable by looking at its internal IP. Neat! But for the purpose of these docs, we don't care about that. The docs recommended a StandaloneVM for simplicity, but I'm actually using an AppVM myself. And since `10.137.0.1/24` and `10.138.0.1/24` are not contiguous, there's no point in trying to unify them in a single subnet declaration. 

Looks like the `10.137.0.1/24` is the correct answer to our question, and that's what we should include in the docs. 

Does the above look accurate to you, @gonzalo-bulnes? If so, do you agree we should tweak that line in this PR? That was the only quibble I had with the content—the rest of the diff looks just swell to my eye.

gonzalo-bulnes commented:
> The question I'm trying to answer is "What is the CIDR subnet encompassing IPs for Qubes VMs?"
:white_check_mark: That's and was my intention too.

Not including the DispVM. Seems all right to me. I can't think of a DispVM that we'd need to connect to _through the network_ (RPC are more likely for passing documents around, am I correct?).

Wouldn't the AppVMs need to be able to reach the name severs though? Those are assigned `10.139.1.1` and `10.139.1.2`. I can look up where I got that from, but I did confirm it, maybe as part of https://qubes-os.discourse.group/t/3870

gonzalo-bulnes commented:
Also, answering your initial question @conorsch, I tested this on R4.1. I don't believe there have been changes in that part of Qubes OS between R4.0 and R4.1.

gonzalo-bulnes commented:
Did a two things at once:
- Updated the branch (rebased)
- Changed the subnet recommendation to `10.137.0.0/24`

@conorsch If you don't feel strongly about `10.137.0.1/24`, I find a little easier on the mind to keep the segments that are variable in a subnet to `0`. That being said I don't have strong feelings about that either.

Also, I'd favor `10.137.0.0/24` over "the value returned by `qvm-prefs sd-staging-base-focal ip`, with `/24` netmask suffix" because we know the gateway will be in the `10.137.0.0/24` range anyway. (The docs are long enough to try and go to the point IMHO.)

gonzalo-bulnes commented:
Thanks for the thoughtful and thorough review @conorsch!

conorsch reviewed (approved):
Love it. Thanks for the in-depth discussion, @gonzalo-bulnes, I learned a lot as part of review here! :rainbow:
